### ISSUE TITLE
[Crash] Cannot exit the game

### ISSUE THREAD

**Issue by asanagisae** [tsunamayo/Starship-EVO#5802]
### Short Description of the issue:

After playing for a while, I am unable to exit the game normally.

I am unsure of reproduction steps. Most actions taken during the session could be considered "normal", at least normal for a survival world. This world is creative. Actions taken include:

- Placing blocks on a ship
- Building handhelds and placing them on item stands
- Using the repair tool to repair interior blocks after accidentally discharging the firearm indoors
- Mining ores from a planet surface
- Having the ship pick up the ores using a collector block
- Refining ores with an arc furnace aboard the ship
- Producing items from the assembler
- Placing batteries on power cell bases
- Charging batteries placed on power cell bases

Hopefully, this report will help prevent future crashes from happening for other players.

### Steps to reproduce the issue:

1. Press Esc, choose "Exit to Desktop".
2. The "Exit to Desktop" button does not respond, the game does not exit.

The ESC menu visibility can be toggled with the ESC key. The mouse cursor continues to function. It appears that the game is not "closed" or stalled. However, the world and all ships are "unloaded" from view.

After encountering this issue, I used Steam to forcibly close the game. When I reloaded the game, it appears that the world reverted to a previous save, taken a few minutes prior.

I had been regularly saving the ship I was working on. The version in the world was an earlier version. When I use a starter block to spawn the blueprint in, this new version is up-to-date with my last blueprint export (as expected).

See the player log for details.

### Environment

- Starship EVO version: 24w20b [default branch] [creative mode]

### Additional information:

This log should be from the session with the crash.
[Player.log](https://github.com/user-attachments/files/15811692/Player.log)

The collector block appears to have picked up some trees and shrubs. I'm not sure if this is intended behavior, or if it affects the save file (as removed trees need to be noted and saved).

![image](https://github.com/tsunamayo/Starship-EVO/assets/10069469/fd451c80-a9e7-40f9-890b-df805ca0bedb)

I checked the size of the character save file. It is 54 MB currently. This is significantly less than the amount of free space on the C:, which is 7 GB.

**Comment by tsunamayo** [comment]
@asanagisae Nope the collector is a bug, I will look into your crash, it seems there is tons of issues in the log file.

**Comment by tsunamayo** [comment]
@asanagisae Hi, could you send me your character save file, at the position you are when you get the crash, it seems to be related to terrain saving. Cheers

**Comment by asanagisae** [comment]
I'm not sure that request makes sense?

When it crashes, it fails to save the world data. Therefore, the world save I attach here will be from an earlier point in the play session and the bug will not manifest. Both times when this happened, I continued the save as normal (from the earlier point) and the crash did not happen in that following session.

If I only move to the crash location and take no other action, then you should have a normal save file that does not reproduce the issue.

Regardless, I can try to reproduce it again and capture it with video, and attach the resulting world save.

**Comment by asanagisae** [comment]
I ran into the bug again when I wasn't expecting it. For this session, I turned on sound and walked around the landing site with my mech to see what the hinges with custom sounds sounded like, all while firing the lasers in every direction except towards my transport ship.

[Player.log](https://github.com/user-attachments/files/15921489/Player.log)
[Mechatro WeGo.zip](https://github.com/user-attachments/files/15921492/Mechatro.WeGo.zip)

When opened the world again to try and reproduce it with video recording software running, the crash did not reoccur.
